src/main/java/org/sefglobal/scholarx/service/MenteeService.java

```diff
@@ -95,9 +95,10 @@ public Mentee approveOrRejectMentee(long menteeId, long profileId, Boolean isApp
             optionalMentee.get().setRejectedBy(mentor);
             optionalMentee.get().setAssignedMentor(null);
         } else if (optionalMentee.get().getState().equals(EnrolmentState.APPROVED)) {
+            optionalMentee.get().setRejectedBy(mentor);
+            optionalMentee.get().setAssignedMentor(null);
             mentor.setNoOfAssignedMentees(mentor.getNoOfAssignedMentees() - 1);
         }
-        optionalMentee.get().setAssignedMentor(mentor);
 
         optionalMentee.get().setState(isApproved?EnrolmentState.APPROVED:EnrolmentState.REJECTED);
         return menteeRepository.save(optionalMentee.get());
```

Comment:
We should add a zero limit into this somehow, because if a mentor only rejects mentees then the count would go down into negatives
this happens because we reset the assigned mentee count to zero moving into the mentee selection phase